src/Client.ts

```diff
@@ -59,49 +60,43 @@ export class Client implements IHasHTTP {
    */
   public static async getMetadata(server: OnmsServer, httpImpl?: IOnmsHTTP, timeout?: number):
     Promise<ServerMetadata> {
-    const opts = new OnmsHTTPOptions(timeout, server.auth, server);
-    opts.headers.accept = 'application/json';
+    let opts = new OnmsHTTPOptions(timeout, server).withHeader('Accept', 'application/json');
     if (!httpImpl) {
       if (!Client.defaultHttp) {
         throw new OnmsError('No default HTTP implementation is configured!');
       }
       httpImpl = Client.defaultHttp;
     }
     if (!timeout && httpImpl && httpImpl.options && httpImpl.options.timeout) {
-      opts.timeout = httpImpl.options.timeout;
+      opts = opts.withTimeout(httpImpl.options.timeout);
```

Comment:
If we use the builder pattern, here we would be conditionally setting on the builder rather than creating a new `OnmsHTTPOptions` via `with`.